File: c++/code/Stuff.cpp
#include "Stuff.hpp"
#include <iostream>
namespace myspace{
    using namespace std;
    Stuff::Stuff(int id,double salary,const string& name):m_name(name),m_salary(salary),m_id(id){

        }

    double Stuff::getSalary() const{
        return this->m_salary;
    }
    void Stuff::setSalary(double salary){
        this->m_salary=salary;
    }
    int Stuff::getId() const{return this->m_id;}
    void Stuff::Show() const
    {
        std::cout<<std::endl<<"this is stuff "+m_id<<std::endl;
        std::cout<<m_id<<std::endl;
    }
    Stuff::~Stuff(){}
}

File: java/test/src/generic/Stuff.java
package generic;

public abstract class Stuff<T> implements Comparable<T> {

	private int m_id;
	private double m_salary;
	protected String m_name;
	
	
	
	public Stuff(){
		m_id=(int)(Math.random()*100);
		m_salary=Math.random()*1000;
		m_name="no name";
		
	}
	
	public Stuff (int id,double salary,String name)
	{
		m_id=id;
		m_name=name;
		m_salary=salary;
	}
	
		
	
	 public String getName()
	 {
		 return m_name;
	 }
	 public double getSalary()
	 {
		 return m_salary;
		 
	 }
	 public int getId()
	 {
		 return m_id;
	 }
	 public String toString()
	 {
		 return "Employee "+m_id+"  name= "+m_name+" salary"+m_salary;
		 
	 }
	public int comparaTo(T other)
	{
		return 0;
	}
	
	
}
